odoo/osv/expression.py

```diff
@@ -580,10 +585,12 @@ def add_join_context(self, model, lhs_col, table_col, link):
     def get_join_conditions(self):
         conditions = []
         alias = self._models[0]._table
+        shortened_alias = alias
         for context in self.join_context:
-            previous_alias = alias
+            previous_alias = shortened_alias
             alias += '__' + context[4]
-            conditions.append('"%s"."%s"="%s"."%s"' % (previous_alias, context[2], alias, context[3]))
+            shortened_alias = _shorten_alias(alias)
+            conditions.append('"%s"."%s"="%s"."%s"' % (previous_alias, context[2], shortened_alias, context[3]))
```

Comment:
IMHO the variable `shortened_alias` has no added value.  You can simplify the loop to
```python
for context in self.join_context:
    previous_alias = alias
    alias = _shorten_alias(alias + '__' + context[4])
    conditions.append('"%s"."%s"="%s"."%s"' % (previous_alias, context[2], alias, context[3]))
```